Mathlib/RingTheory/AdicCompletion/Basic.lean

```diff
@@ -228,6 +261,18 @@ def incl : AdicCompletion I M →ₗ[R] (∀ n, M ⧸ (I ^ n • ⊤ : Submodule
   map_add' _ _ := rfl
   map_smul' _ _ := rfl
 
+variable {I M}
+
+@[simp, norm_cast]
+lemma val_sum {ι : Type*} (s : Finset ι) (f : ι → AdicCompletion I M) :
+    (∑ i ∈ s, f i).val = ∑ i ∈ s, (f i).val := by
+  simp_rw [← funext (incl_apply _ _ _), map_sum]
+
+lemma val_sum_apply {ι : Type*} (s : Finset ι) (f : ι → AdicCompletion I M) (n : ℕ) :
+    (∑ i ∈ s, f i).val n = ∑ i ∈ s, (f i).val n := by simp
```

Comment:
If .val were an AddMonoidHom you'd get all this stuff for free, right?

Reply:
This one no, since here `val` is applied to two arguments. The one above, yes, but you can already get it for "free" with `map_sum (incl _) _ _` or similar (where `incl` is the `AddMonoidHom` you're asking for)